frontend/src/components/checkout/place-order/order-itmes/ItemsList.tsx

```diff
@@ -30,44 +29,57 @@ export default function ItemsList() {
-                        </Card>
-                        <Typography>{ productId }</Typography>
-                        <Typography>{ price }$</Typography>
-                        <Button
-                            onClick={ handleQuantityDecrement(productId) }
-                            disabled={ quantity === 1 }
-                        >
-                            <RemoveIcon />
-                        </Button>
-                        <Typography>{ quantity }</Typography>
+                }: product) => {
+
+                    const disabledButtonLogic =
+                        quantity === productLimit ||
+                        quantity === 6 ||
+                        quantity >= countInStock;
+
+                    return (
+                        <Card id="card-container" sx={ cardContainer } key={ productId }>
+                            <Card id="image-container">
+                                <img id="image" style={ itemImage } src={ image } alt="items-img" />
+                            </Card>
+                            <Typography id="product-id">{ productId }</Typography>
+                            <Typography id="product-price">{ price }$</Typography>
+                            <Button
+                                id="decrement-button"
+                                onClick={ handleQuantityDecrement(productId) }
+                                disabled={ quantity === 1 }
+                            >
+                                <RemoveIcon />
+                            </Button>
+                            <Typography id="product-quantity">{ quantity }</Typography>
 
-                        <Button
-                            onClick={ handleQuantityIncrement(
-                                productId,
-                                productLimit,
-                                quantity,
-                                countInStock,
-                            ) }
-                            disabled={ quantity === productLimit || quantity === 6 || quantity >= countInStock }
-                        >
-                            <AddIcon />
-                        </Button>
-                        { window.location.pathname === "/cart" && (
                             <Button
-                                onClick={ handleProductDelete(productId) }
-                                variant="contained"
-                                startIcon={ <DeleteIcon /> }
-                                sx={ deleteButton }
+                                id="increment-button"
+                                onClick={ handleQuantityIncrement(
+                                    {
+                                        productId,
+                                        productLimit,
+                                        quantity,
+                                        countInStock
+                                    }
+                                ) }
```

Comment:
The `onClick` event should be assigned a function reference rather than invoking the function immediately.

```
onClick={() => handleQuantityIncrement({
    productId,
    productLimit,
    quantity,
    countInStock
})}
```

Reply:
True, but in the function handleQuantityIncrement is returning another function which will be invoked when the button is clicked.
`const handleQuantityIncrement =
        ({
            productId,
            productLimit,
            quantity,
            countInStock,
        }: HandleQuantityIncrement) =>
        here.....
            () => {
                const updatedCartItems = cartItems.map((item) =>
                    item.productId === productId
                        ? item.quantity < productLimit && item.quantity < countInStock // check if quantity is less than productLimit and countInStock
                            ? { ...item, quantity: item.quantity + 1 }
                            : item
                        : item
                );

                if (quantity >= countInStock) {
                    toast.error(`This product is out of stock`);
                } else if (quantity >= 6) {
                    toast.error(`You have reached the limit for this product`);
                } else if (
                    JSON.stringify(updatedCartItems) === JSON.stringify(cartItems)
                ) {
                    toast.error(`You cannot add more than ${productLimit} items`);
                } else {
                    setCartItems(updatedCartItems);
                }
            };`

Reply:
oh okay! you're using a higher-order function approach, that's fine 👌🏼